### PR TITLE
[ContactStructuralMechanicsApplication] Corrections in predict of contact stress in contact mechanics

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR does some corrections in the prediction of the contact stress. Additionally improves slightly the simplified semi-smooth method

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py [hunk 1]
```diff
@@ -183,6 +184,17 @@ def SolveSolutionStep(self):
...
 
+    def ComputeDeltaTime(self):
+        delta_time = self.settings["time_stepping"]["time_step"].GetDouble()
+        if self.contact_settings["inner_loop_adaptive"].GetBool():
+            process_info = self.GetComputingModelPart().ProcessInfo
+            if process_info.Has(CSMA.INNER_LOOP_ITERATION):
+                inner_iterations = process_info[CSMA.INNER_LOOP_ITERATION]
+                if inner_iterations > 1:
+                    delta_time = delta_time/float(inner_iterations)
+                    self.print_on_rank_zero("::[Contact Mechanical Static Solver]:: ", "Advancing with a reduced delta time of ", delta_time)
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:195]: this entire function is copy-pasted
why don't you move such code to a different file or the baseclass?

**Review by RiccardoRossi — APPROVED** [review]
I am approving so that you can make progress, however please take into account the comment of @philbucher in the future

**Comment by loumalouomega** [comment]
Thanks, I will, but certainly I don't know how to fix the issue,being the solvers deriving from different classes but with common utilities. The only thing I can think is in creating some utilities with many arguments

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py (hunk 1 above)
Comment (philbucher) [reply applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:195]: Ok I understand

Still there is a typo in this line :)

Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:195]: OK, I will fix in a future PR

**Pull request merged**